full_name.rb

```diff
@@ -0,0 +1,22 @@
+
+
+def greetings
+    greet_name = []
+
+    print "What is your first name? "
+    first_name = gets.chomp
+    greet_name.push(first_name)
+
+    print "What is your middle name? "
+    middle_name = gets.chomp
+    greet_name.push(middle_name)
+
+    print "What is your last name? "
+    last_name = gets.chomp
+    greet_name.push(last_name)
+
+    # greet_name.each {|name| puts "We are happy to greet you " + name}
+    puts "We are happy to greet you #{greet_name.join(' ')}!"
+end
+
+greetings()
```

Comment:
This code is good. It is important to have some sort of validation with the inputs (I would say mainly in first name and last name) because, if we leave them blank, this is the final message:
`We are happy to greet you   !`
 
We can add some sort of conditional. If the user doesn't enter anything either first or last name, the program returns a message saying something like "This field is required". This is not mandatory to do, however, it is a good practice always making "full proof" programs.